charts/server/templates/pod-monitor.yaml

```diff
@@ -1,16 +1,19 @@
-{{- if .Values.metrics.enabled }}
+{{- if and .Values.metrics.enabled .Values.prometheus.podmonitor.enabled ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" ) }}
```

Comment:
this is a breaking change (that podmonitor must be enabled either and/or Capabilities-Check) ... ala semver